src/app/Common/Caching/Cache.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Caching.Interfaces;
+using Common.Settings;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Common.Caching;
+public class Cache : ICache
+{
+    private readonly IDistributedCache _cache;
+    private readonly ILogger<Cache> _logger;
+    private readonly CacheSettings _cacheSettings;
+
+    public Cache(
+        IDistributedCache cache,
+        ILogger<Cache> logger,
+        IOptions<CacheSettings> cacheSettings)
+    {
+        _cache = cache;
+        _logger = logger;
+        _cacheSettings = cacheSettings.Value;
+    }
+
+    public async Task<T> GetAsync<T>(string key, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var json = await _cache.GetAsync(key, cancellationToken);
+
+            return json != null
+                ? JsonSerializer.Deserialize<T>(json)
+                : default;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogWarning(ex,
+                "Cache read error ({@ConnectionString}). Key: {@Key}",
```

Comment:
Not sure if it's a good idea to include connection string into logs, since it can contain credentials.